### ISSUE TITLE
Uninstalling and reinstalling broke the app

### ISSUE THREAD

**Issue by efi** [termux/termux-app#69]
After removing the previous version and then installing 0.31 the following occurs when starting the app:
```console
exec("/data/data/com.termux/files/usr/bin/bash/"):
 No such file or directory

[Process completed with code 1 - press Enter to c
lose]
```
Also re-installing 0.29 does not fix it.

Samsung S5 neo with Android 5.1.1

**Comment by fornwall** [comment]
Strange! Could you try uninstalling and re-installing yet again to see if the problem remains?

Do you hqve a custom rom? From where do you install your app (e.g. google play or f-droid)?

**Comment by efi** [comment]
Hi. Yeah, this is really strange. Unfortunately the problem persists with each consecutive re-installation attempt.

I have the standard OS (plain Android, slightly modified by Samsung), unrooted (this is why I can't even look into /data/data).

I think I installed the (then working) old version through F-Droid. Then I removed it by hand and installed the 0.31 via Play Store (since I could not see that version in F-Droid - seems I just forgot to update the packages).

Then when starting, after a (relatively short) phase of "installing system" or something like that, it gives the error message.

**Comment by fornwall** [comment]
If it's possible, you can try running a failsafe shell by pulling out the drawer menu from the left, long pressing the `NEW SESSION` button and pressing the `FAILSAFE` button in the resulting dialog.

If you get a working shell there you can examine the folder structure and perhaps trying to remove everything `rm -Rf $PREFIX $HOME` and exit the shell to force a clean re-install next launch of the app.

Another thing to ensure could be to uninstall all Termux add-ons (such as Termux:Styling, Termux:API, etc) as well as the Termux main app itself before installing again, to see if that ensures a clean install.

I'm closing this issue as I'm trying to clean up issues a bit, just re-open the issue after you have tried the above and need more help!

**Comment by efi** [comment]
Reopening...

Thank you for the advice. I did not know the failsafe option in advance. Thanks to that now I found out that the bash file is actually there and has proper permissions set.

Deleting the folders you mentioned (and in a next step also everything in com.termux) did not solve the problem - even with de- and reinstallations. I do not use add-ons (yet) an so this cannot be the source of the problem either.

Maybe the error message is just misleading? Could it be something with the 32/64 bit architecture like mentioned here: http://unix.stackexchange.com/questions/13391/getting-not-found-message-when-running-a-32-bit-binary-on-a-64-bit-system/13409#13409 ?

**Comment by fornwall** [comment]
@efi Two things to try out in the failsafe shell:

1. What is the output of `uname -a`?
2. Can you try execute `/data/data/com.termux/files/usr/bin/dash` and see if that works?

I just got another report about in the Google Play store about Termux not working with Samsung Galaxy S5 neo, so it could be something with this device.

**Comment by efi** [comment]
1.  ```sh: uname not found```
2.  ```sh: /data/data/com.termux/files/usr/bin/dash: No such file or directory```
but it can be found and opened (e.g. with ```cat```)

Yeah, the S5 neo is a bit special... Could have something to do with the "exotic" Exynos 7580 Processor and how it interacts with the system.

**Comment by fornwall** [comment]
@efi Thanks, it could very well be due to the cpu! It seems that Exynos 7580 is 64-bit capable, but perhaps the S5 neo system doesn't have 64-bit support built in?

Termux could then be confused and try to install 64-bit binaries due to the cpu supporting it, but the system linker does not support running it and gives the not found error just as you linked earlier.

Can you run the http://fornwall.net/termux-debug.apk app and see the output from that? It's a single textview showing something like `os.arch=...`? Using `System.getProperty("os.arch")` is how Termux decides which arch to install the initial binaries for.

Also, from the failsafe shell:

1. Can you check if the file `/system/bin/linker64` exists (e.g. by running `ls /system/bin/linker64`)?
2. What is the output of `cat /proc/cpu`?
3. What is the output of `/system/bin/getprop  | grep cpu`?

Thanks for all your help here!

**Comment by efi** [comment]
You debug program recognizes the following arch: ```aarch64```

* 1. there's only a ```linker```
* 2. there is no ```/proc/cpu``` only ```cpuinfo``` which says:
```
Processor       : AArch64 Processor Rev 3 (aarch64)
processor       : 0
processor       : 1
processor       : 2
processor       : 3
Features        : fp asimd aes pmull sha1 sha2 crc32 wp half thumb fastmult vfp edsp neon vfpv3 tlsi vfpv4 idiva idivt
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd03
CPU revision    : 3

Hardware : SAMSUNG Exynos7580
```
* 3. 
```
[ro.product.cpu.abi2]: [armeabi]
[ro.product.cpu.abi]: [armeabi-v7a]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: []
[ro.product.cpu.abilist]: [armeabi-v7a,armeabi]
```
(all copied by hand, so may contain typos)

**Comment by fornwall** [comment]
@efi Thanks a lot, this is enough for me to create a fix! I'll ping you shortly when I've released an update.

**Comment by fornwall** [comment]
@efi This should now be fixed in the v0.33 of the Termux app which should be available as an update shortly - try uninstalling and then reinstalling this version and let me know how it works!

**Comment by efi** [comment]
Hi Fredrik,

thanks for the fixes! Works fine again.

Still I'm a bit disappointed in getting only a 32 bit core system when the Processor and all is advertised as 64 bit. Maybe Samsung will fix this with the upcoming Android 6 upgrade.
